sdk/src/signature.rs

```diff
@@ -396,10 +397,37 @@ pub fn keypair_from_seed(seed: &[u8]) -> Result<Keypair, Box<dyn error::Error>>
     Ok(Keypair(dalek_keypair))
 }
 
-pub fn keypair_from_seed_phrase_and_passphrase(
+/// Generates a Keypair using Bip32 Hierarchical Derivation if derivation-path is provided;
+/// otherwise builds standard Keypair using the seed as SecretKey
+pub fn keypair_from_seed_and_derivation_path(
+    seed: &[u8],
+    derivation_path: Option<DerivationPath>,
+) -> Result<Keypair, Box<dyn error::Error>> {
+    if let Some(derivation_path) = derivation_path {
+        bip32_derived_keypair(seed, derivation_path).map_err(|err| err.to_string().into())
+    } else {
+        keypair_from_seed(seed)
+    }
```

Comment:
Here is the `if` clause I mentioned to retain compatibility with current keypair creation. Worth considering if we need to do this. We could transition all creation-from-seed to `bip32_derived_keypair` with derivation_path of length 0